src/test/java/brave/opentracing/BraveSpanTest.java

```diff
@@ -145,4 +145,24 @@ public void clear() {
     assertThat(zipkin.spanStore().getDependencies(System.currentTimeMillis(), null))
         .containsExactly(DependencyLink.create("tracer", "tracer2", 1L));
   }
+
+  @Test public void testNotSampled_spanBuilder_parentNull() {
+    tracer.buildSpan("foo")
+            .withTag(Tags.SAMPLING_PRIORITY.getKey(), 0)
+            .start().finish();
+
+    assertThat(zipkin.spanStore().getRawTraces()).isEmpty();
+  }
+
+  @Test public void testNotSampled_spanBuilder_parentNotNull() {
```

Comment:
ex this isn't a supported trace as in B3 and existing zipkin infra, when you are told don't sample, you're supposed to not sample.